src/Pos/Update/Poll/Trans.hs

```diff
@@ -179,14 +186,14 @@ instance MonadPollRead m =>
     addConfirmedProposal cps =
         PollT $ pmConfirmedPropsL %= MM.insert (cpsSoftwareVersion cps) cps
     delConfirmedProposal sv = PollT $ pmConfirmedPropsL %= MM.delete sv
-    addActiveProposal ps =
+    addActiveProposal ps = do
+        let up@UpdateProposal{upSoftwareVersion = sv, ..} = psProposal ps
+            upId = hash up
+            appName = svAppName sv
+        whenNothingM_ (getProposal upId) $ -- if not exist such proposal
```

Comment:
Ok, now it was my turn to forget that `addActiveProposal` can also override existing one.